chat.go

```diff
@@ -54,23 +54,23 @@ type ChatCompletionRequest struct {
 	FrequencyPenalty float32                 `json:"frequency_penalty,omitempty"`
 	LogitBias        map[string]int          `json:"logit_bias,omitempty"`
 	User             string                  `json:"user,omitempty"`
-	Functions        []*FunctionDefine       `json:"functions,omitempty"`
-	FunctionCall     string                  `json:"function_call,omitempty"`
+	Functions        []FunctionDefinition    `json:"functions,omitempty"`
+	FunctionCall     any                     `json:"function_call,omitempty"`
 }
 
-type FunctionDefine struct {
+type FunctionDefinition struct {
 	Name        string `json:"name"`
 	Description string `json:"description,omitempty"`
-	// it's required in function call
-	Parameters *FunctionParams `json:"parameters"`
+	// Parameters is an object describing the function.
+	// You can pass a raw byte array describing the schema,
+	// or you can pass in a struct which serializes to the proper JSONSchema.
+	// The JSONSchemaDefinition struct is provided for convenience, but you should
+	// consider another specialized library for more complex schemas.
+	Parameters any `json:"parameters"`
 }
 
-type FunctionParams struct {
-	// the Type must be JSONSchemaTypeObject
-	Type       JSONSchemaType               `json:"type"`
-	Properties map[string]*JSONSchemaDefine `json:"properties,omitempty"`
-	Required   []string                     `json:"required,omitempty"`
-}
+// Deprecated: use FunctionDefinition instead.
+type FunctionDefine FunctionDefinition
```

Comment:
This is a defined type; cannot use `FunctionDefine` for `Functions` fields. Compatibility can be maintained by using a type alias as follows:

`type FunctionDefine = FunctionDefinition`